### PR TITLE
Add missing operator and assignment expressions

### PR DESCRIPTION
**Pull request by techouse**
I have recently struggled to clean up a package that I maintain using `Expression` operators instead of using `Code` blocks full of strings, i.e. `Code('$a ^ $b')` and have discovered that quite a few operators are missing from the `Expression` class. So I've added all the missing operators and assignments found on https://dart.dev/language/operators.

- added operators

  | method | generates |
  | :--- | :---: |
  | `Expression.operatorIntDivide` | `this ~/ other` |
  | `Expression.operatorUnaryPrefixIncrement` | `++this` |
  | `Expression.operatorUnaryPostfixIncrement` | `this++` |
  | `Expression.operatorUnaryMinus` | `-this` |
  | `Expression.operatorUnaryPrefixDecrement` | `--this` |
  | `Expression.operatorUnaryPostfixDecrement` | `this--` |
  | `Expression.operatorBitwiseAnd` | `this & other` |
  | `Expression.operatorBitwiseOr` | `this \| other` |
  | `Expression.operatorBitwiseXor` | `this ^ other` |
  | `Expression.operatorUnaryBitwiseComplement` | `~this` |
  | `Expression.operatorShiftLeft` | `this << other` |
  | `Expression.operatorShiftRight` | `this >> other` |
  | `Expression.operatorShiftRightUnsigned` | `this >>> other` |

- added assignments

  | method | generates |
  | :--- | :---: |
  | `Expression.addAssign` | `this += other` |
  | `Expression.subtractAssign` | `this -= other` |
  | `Expression.multiplyAssign` | `this *= other` |
  | `Expression.divideAssign` | `this /= other` |
  | `Expression.intDivideAssign` | `this ~/= other` |
  | `Expression.euclideanModuloAssign` | `this %= other` |
  | `Expression.shiftLeftAssign` | `this <<= other` |
  | `Expression.shiftRightAssign` | `this >>= other` |
  | `Expression.shiftRightUnsignedAssign` | `this >>>= other` |
  | `Expression.bitwiseAndAssign` | `this &= other` |
  | `Expression.bitwiseXorAssign` | `this ^= other` |
  | `Expression.bitwiseOrAssign` | `this \|= other` |

- deprecated the misnamed `Expression.operatorSubstract` method and created a correctly named `Expression.operatorSubtract` method.

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

### CODE REVIEW

**Review by natebosch — APPROVED** [review]
LGTM.

File: lib/src/specs/expression.dart [hunk 1]
```diff
@@ -189,10 +195,106 @@ abstract class Expression implements Spec {
...
+  /// Returns the result of `this` `>>>` [other].
+  Expression operatorShiftRightUnsigned(Expression other) =>
+      BinaryExpression._(expression, other, '>>>');
+
   /// Return `{this} = {other}`.
   Expression assign(Expression other) =>
       BinaryExpression._(this, other, '=', isConst: isConst);
 
+  /// Return `this` += [other].
+  Expression assignAdd(Expression other) =>
```
Comment (natebosch) [inline lib/src/specs/expression.dart:251]: @srawlins - can you double check that the naming patterns LGTY? Are there any existing (short) names for these compound assignment operators that you think we should switch to?

Comment (techouse) [reply lib/src/specs/expression.dart:251]: @natebosch @srawlins I wasn't too sure whether to go with
```dart
Expression assignAdd(Expression other);
```
or
```dart
Expression addAssign(Expression other);
```

However looking at https://processing.org/reference/addassign.html I see that they refer to `+=` as `add assign`, so naming the operators from left to right.

I can change the names if desired.

Comment (techouse) [reply lib/src/specs/expression.dart:251]: Looking at Rust https://doc.rust-lang.org/std/ops/trait.AddAssign.html I probably should rename them
| from | to | 
| --- | --- |
| `Expression.assignAdd` | `Expression.addAssign` |
| `Expression.assignSubtract` | `Expression.subtractAssign` |
| `Expression.assignMultiply` | `Expression.multiplyAssign` |
| `Expression.assignDivide` | `Expression.divideAssign` |
| `Expression.assignIntDivide` | `Expression.intDivideAssign` |
| `Expression.assignEuclideanModulo` | `Expression.euclideanModuloAssign` |
| `Expression.assignShiftLeft` | `Expression.shiftLeftAssign` |
| `Expression.assignShiftRight` | `Expression.shiftRightAssign` |
| `Expression.assignShiftRightUnsigned` | `Expression.shiftRightUnsignedAssign` |
| `Expression.assignBitwiseAnd` | `Expression.bitwiseAndAssign` |
| `Expression.assignBitwiseXor` | `Expression.bitwiseXorAssign` |
| `Expression.assignBitwiseOr` | `Expression.bitwiseOrAssign` |

Comment (techouse) [reply lib/src/specs/expression.dart]: ^^ done

Comment (srawlins) [reply lib/src/specs/expression.dart]: I don't think analyzer APIs have anything to add here. We call assignments like `+=` a "compound assignment", but don't have a specific term for `+=` vs `-=`. Those symbols are distinguished by the TokenTypes, which also don't really make an attempt to name anything, beyond exactly what the characters are. For example [PLUS_EQ](https://pub.dev/documentation/analyzer/latest/dart_ast_token/TokenType/PLUS_EQ-constant.html), and [TILDE_SLASH_EQ](https://pub.dev/documentation/analyzer/latest/dart_ast_token/TokenType/TILDE_SLASH_EQ-constant.html).

**Review by natebosch — APPROVED** [review]

**Pull request merged**